verilog/tools/kythe/indexing_facts_tree_extractor.cc

```diff
@@ -138,15 +140,50 @@ void IndexingFactsTreeExtractor::ExtractModule(
     Visit(module_item_list);
   }
 
-  facts_tree_context_.back()->NewChild(module_node);
+  facts_tree_context_.top().NewChild(module_node);
 }
 
 void IndexingFactsTreeExtractor::ExtractModuleHeader(
     const verible::SyntaxTreeNode& node) {
   const verible::TokenInfo& module_name_token = GetModuleNameToken(node);
   const Anchor module_name_anchor(module_name_token, context_.base);
 
-  facts_tree_context_.back()->Value().AppendAnchor(module_name_anchor);
+  facts_tree_context_.top().Value().AppendAnchor(module_name_anchor);
+
+  const verible::SyntaxTreeNode* port_list = GetModulePortDeclarationList(node);
+
+  if (port_list != nullptr) {
+    std::vector<verible::TreeSearchMatch> port_names =
+        FindAllUnqualifiedIds(*port_list);
```

Comment:
This might match too many occurrences of unqualified IDs, which can appear in the types referenced by declarations as well.

Suggestion:
From `ModulePortDeclarationList`, extract `kPortDeclaration`s (could use `FindAllModulePortDeclarations` function from CST/port.h).
From each port declaration, `GetIdentifierFromModulePortDeclaration` (CST/port.h).

Reply:
I did that. But what about this case: `module m(a, b);`

Reply:
In this "non-ANSI" style, one would expect to see a full declaration of `a` and `b` in the module body.  Like this:

```systemverilog
module m(a, b);
  input a;
  output b;
endmodule
```

I would consider `input a` the actual definition, and `a` is just a reference.
Make sure you can gracefully handle the case where user makes an error and the ports don't match up.